noq_django/rest_api/api/test/test_volunteer_booking.py

```diff
@@ -0,0 +1,108 @@
+import sys
+sys.path.append("....backend") 
+import json
+from datetime import datetime, timedelta
+from django.test import TestCase
+from django.core import mail
+from django.contrib.auth.models import User, Group
+from backend.models import (
+    Host, Client, Product, Region, Booking, BookingStatus, Available
+)
+from django.test import Client as TestClient
+from ..volunteer_api import router  
+
+class TestVolunteerApi(TestCase):
+    # Constants for test data
+    VOLUNTEER_USERNAME = "volunteer@test.com"
+    PASSWORD = "TestPassword123!"
+    CLIENT_EMAIL = "guest@test.com"
+
+    def setUp(self):
+        # Create a volunteer user and set up initial data
+        self.volunteer_user = User.objects.create_user(
+            username=self.VOLUNTEER_USERNAME, password=self.PASSWORD
+        )
+        volunteer_group, _ = Group.objects.get_or_create(name="volunteer")
+        self.volunteer_user.groups.add(volunteer_group)
+
+        # Set up test client for making requests
+        self.client = TestClient(router)
+        self.client.login(username=self.VOLUNTEER_USERNAME, password=self.PASSWORD)
+
+        # Create a region, host, product, and client data
+        self.region = Region.objects.create(name="Stockholm")
+        self.host = Host.objects.create(name="TestHost", city="Stockholm", region=self.region)
+        self.product = Product.objects.create(name="TestRoom", total_places=5, host=self.host, type="room")
+        self.client_user = Client.objects.create(
+            first_name="Guest",
+            last_name="User",
+            email=self.CLIENT_EMAIL,
+            region=self.region,
+            user=self.volunteer_user,
+            gender="M"  # Add gender here (e.g., "M" or "F")
+    )
+
+    def test_request_booking(self):
+        # Define booking dates
+        start_date = datetime.now().date()
+        end_date = start_date + timedelta(days=2)
+
+        # Prepare booking data
+        booking_data = {
+            "product_id": self.product.id,
+            "user_id": self.client_user.id,
```

Comment:
As volunteer it should be possible to make a booking with just unokod, even if user doesn't have a login and no client data. Client data should probably be created first time someone makes a booking with new unokod.

Reply:
There is another ticket for booking, we can leave the UNOKOD for that ticket.